kimchi/wasm/src/srs.rs

```diff
@@ -213,35 +215,171 @@ macro_rules! impl_srs {
 //
 
 pub mod fp {
+    use std::collections::HashMap;
+
     use super::*;
-    use crate::arkworks::{WasmGVesta, WasmPastaFp};
+    use crate::arkworks::{WasmGVesta as WasmG, WasmPastaFp};
     use crate::poly_comm::vesta::WasmFpPolyComm as WasmPolyComm;
-    use mina_curves::pasta::{Fp, Vesta};
-
-    impl_srs!(
-        caml_fp_srs,
-        WasmPastaFp,
-        WasmGVesta,
-        Fp,
-        Vesta,
-        WasmPolyComm,
-        Fp
-    );
+    use mina_curves::pasta::{Fp, Vesta as G};
+    use poly_commitment::PolyComm;
+
+    impl_srs!(caml_fp_srs, WasmPastaFp, WasmG, Fp, G, WasmPolyComm, Fp);
+
+    #[wasm_bindgen]
```

Comment:
It seems it is duplicated code that can be parametrized by the curve (I might be wrong by reading only GitHub panel). Can it be moved into the macro `impl_srs!`?

Reply:
you're right, could be in the macro. I think it's easier to maintain outside the macro, see my slack thread here: https://o1-labs.slack.com/archives/C01UNQH8XEZ/p1698141695847899

Reply:
Mmh, I see. I understand your point, and I do agree (I hate macros...). However, IMO, it is good to stay with one convention, and maybe initiate a refactorization after.
Not a blocking comment for this PR, only an opinion.